### PR TITLE
Windows build and functionality

### PR DESCRIPTION
**Pull request by vvuk**
Adds support for building on Windows, and fixes some code issues in Skia that prevent building on gcc.

[Review on Reviewable](https://reviewable.io/reviews/servo/skia/74)

### CODE REVIEW

File: makefile.cargo [hunk 1]
```diff
@@ -626,6 +631,61 @@ endif
...
+		SkOSFile_win.cpp \
+		SkOSFile_stdio.cpp \
+		SkTLS_win.cpp \
+	)
+
+#		SkTypeface_win_dw.cpp \
+#		SkFontHost_win.cpp \
+#		SkFontMgr_win_dw.cpp \
+#		SkFontMgr_default_dw.cpp \
+#		SkScalerContext_win_dw.cpp \
```
Comment (mrobinson) [inline makefile.cargo]: Instead of leaving commented lines, I think it would be better to remove them.

Comment (Ms2ger) [inline makefile.cargo]: And  note that comments that end with `\` will eat the next line, so definitely never leave any of those.

**Comment by vvuk** [comment]
Okay, commits are split out now.

For the "build fixes" commit -- I need to not do a static link into the rlib here because -lskia ends up needing to go on the link line once after rust-azure.  With static, that doesn't happen, as all the rlibs are passed to the linker before any -l libs.  For simplicity, we could just do non-static linking for all platforms here.

The uintptr_t fix is already in skia in mozilla-central; not sure if it's in upstream or not.

The ## pasting fix is not in mozilla-central; not sure why it's not a problem there when being built with gcc.

And the final SK_INCLUDE_FREETYPE temporary thing is an addition that will go away once we actually switch Servo to using DirectWrite on windows.

**Comment by vvuk** [comment]
Anything I can do to speed up getting this merged in?

**Comment by Manishearth** [comment]
r? @glennw 

cc @larsbergstrom

**Comment by larsbergstrom** [comment]
I'm fine with everything except for:
- My one comment on the linking
- The second patch (on gl context / rasterization stuff), which it would be great if @glennw  could look at.

**Comment by glennw** [comment]
Just a few minor comments on that commit, mostly looks good to me!

**Comment by vvuk** [comment]
Okay, fixed the build.rs linking stuff -- we can do a static link of skia just fine.  Depends on https://github.com/servo/rust-azure/pull/205 though (without it, there are library ordering issues).

**Comment by metajack** [comment]
re-r? @glennw

**Comment by glennw** [comment]
The gl code still looks fine to me - passing back to @larsbergstrom for the other commits.

**Comment by larsbergstrom** [comment]
Reviewed 7 of 15 files at r1, 6 of 8 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/skia/74)

**Comment by larsbergstrom** [comment]
Gah! Defeated by the libc madness...

**Comment by jdm** [comment]
I'm having a hard time figuring out what in the dependency chain relies on a pinned version of glutin :<

**Comment by larsbergstrom** [comment]
@bors-servo: retry

Let's see if enough of the libcpocalypse has been sorted out for this to land or if we need to get a few more things.

**Comment by jdm** [comment]
```
/home/travis/.cargo/registry/src/github.com-0a35038f75765ae4/glutin-0.3.7/src/api/osmesa/mod.rs:82:63: 83:69 error: mismatched types:
/home/travis/.cargo/registry/src/github.com-0a35038f75765ae4/glutin-0.3.7/src/api/x11/window.rs:517:38: 517:63 error: mismatched
```

**Comment by larsbergstrom** [comment]
Problem is that skia directly relies on glutin 0.3.7 (via a "0.3.5"). Changing this to "0.4" works!

Though I'm a little worried because we are using a glutin from crates.io *and* the servo version in servo proper?

**Comment by larsbergstrom** [comment]
Oh, actually, it looks like the glutin line was added here and should just be removed from this PR.

@vvuk Why was that line added? Is it required on Windows?

**Comment by vvuk** [comment]
src/gl_content_wgl.rs is implemented in terms of glutin on Windows, so glutin is required.  There's no specific dependency on a version of glutin... also in theory mixing mutliple versions shouldn't be an issue, since the glutin used in skia is kept purely to internal skia code.

**Comment by larsbergstrom** [comment]
Merged in https://github.com/servo/skia/pull/82 instead.

**Pull request closed without merging**